src/commands/step-functions/instrument.ts

```diff
@@ -0,0 +1,232 @@
+      const hasEnvTag = listStepFunctionTagsResponse?.tags?.some((tag) => tag.key === 'env')
+      if (!hasEnvTag && typeof this.environment === 'string') {
+        stepFunctionTagsToAdd.push({
+          key: 'env',
+          value: this.environment,
+        })
+      } else if (!hasEnvTag && this.environment === undefined) {
+        this.context.stdout.write('[Error] --env is required when a Step Function has no env tag\n')
+
+        return 1
+      }
+
+      if (
+        !listStepFunctionTagsResponse?.tags?.some((tag) => tag.key === 'service') &&
+        typeof this.service === 'string'
+      ) {
+        stepFunctionTagsToAdd.push({
+          key: 'service',
+          value: this.service,
+        })
+      }
+
+      // set version tag if it changed
+      if (
+        !listStepFunctionTagsResponse?.tags?.some(
+          (tag) => tag.key === TAG_VERSION_NAME && tag.value === `v${cliVersion}`
+        )
+      ) {
+        stepFunctionTagsToAdd.push({key: TAG_VERSION_NAME, value: `v${cliVersion}`})
+      }
+
+      if (stepFunctionTagsToAdd.length > 0) {
+        const tagStepFunctionRequest = tagStepFunction(stepFunctionsClient, stepFunctionArn, stepFunctionTagsToAdd)
+        requestsByStepFunction[stepFunctionArn].push(tagStepFunctionRequest)
+      }
+
+      const subscriptionFilterName = buildSubscriptionFilterName(stepFunction.name)
+
+      const logLevel = stepFunction.loggingConfiguration?.level
+      if (logLevel === 'OFF') {
+        // if step function logging is disabled, create a log group, subscribe the forwarder to it, and enable step function logging to the created log group
+        const logGroupName = buildLogGroupName(stepFunction.name, this.environment)
+        const createLogGroupRequest = createLogGroup(cloudWatchLogsClient, logGroupName)
+        requestsByStepFunction[stepFunctionArn].push(createLogGroupRequest)
+
+        const putSubscriptionFilterRequest = putSubscriptionFilter(
+          cloudWatchLogsClient,
+          this.forwarderArn,
+          subscriptionFilterName,
+          logGroupName
+        )
+        requestsByStepFunction[stepFunctionArn].push(putSubscriptionFilterRequest)
+
+        const logGroupArn = buildArn(
+          arnObject.partition,
+          'logs',
+          arnObject.region,
+          arnObject.accountId,
+          'log-group',
+          `${logGroupName}:*`
+        )
+
+        // IAM policy on step function role should already include log permissions
+        const enableStepFunctionLogsRequest = enableStepFunctionLogs(stepFunctionsClient, stepFunction, logGroupArn)
+        requestsByStepFunction[stepFunctionArn].push(enableStepFunctionLogsRequest)
+      } else {
+        // if step function logging is enabled, subscribe the forwarder to the log group in the step function logging config
+        const logGroupArn = getStepFunctionLogGroupArn(stepFunction)
```

Comment:
If returning empty string here, we should probably `return 1` to stop early?

Reply:
Agreed! I updated `getStepFunctionLogGroupArn` to return `undefined` and for the command to `return 1` if `undefined` is returned.